exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillMergeProjectRule.java

```diff
@@ -48,6 +50,12 @@ public boolean matches(RelOptRuleCall call) {
     Project topProject = call.rel(0);
     Project bottomProject = call.rel(1);
 
+    // Make sure both projects be LogicalProject.
+    if (topProject.getTraitSet().getTrait(ConventionTraitDef.INSTANCE) != Convention.NONE ||
```

Comment:
You are right that the current Drill specific MergeProjectRule extends from Calcite's version and the only difference is to check complex output expression. Therefore, it make senses the currently Drill MergeProjectRule operates on Calcite logical convention only. 

Yes, in the future, we would consider allowing merging 2 DrillProjectRels. That should be another instance of rule, which operates on DRILL_LOGICAL convention only.